pkg/applicationserver/io/pubsub/pubsub.go

```diff
@@ -279,37 +290,31 @@ func (ps *PubSub) start(ctx context.Context, pb *ttnpb.ApplicationPubSub) (err e
 		case <-ctx.Done():
 		}
 	}()
-	format, ok := formats[pb.Format]
-	if !ok {
+	var ok bool
+	if i.format, ok = formats[pb.Format]; !ok {
 		return errFormatNotFound.WithAttributes("format", pb.Format)
 	}
-	i.format = format
+
 	go i.handleUp(ctx)
 	i.startHandleDown(ctx)
-	logger.Info("Started")
+	logger.Info("Pub/sub started")
 	registerIntegrationStart(ctx, i)
+
 	<-ctx.Done()
-	if err := ctx.Err(); errors.IsCanceled(err) {
-		logger.Info("Integration canceled")
-		i.conn.Shutdown(ctx)
```

Comment:
Shutting down the provider connection only if the error type is `cancelled` is fundamentally wrong. We should always cleanup the connection.